File: backend/PortalTransparenciaDeps/PortalTransparenciaDeps.Web/Models/SancoesAggregate/CnepModel.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace PortalTransparenciaDeps.Web.Models.SancoesAggregate
{
    public class CnepModel
    {
        [JsonPropertyName("abrangenciaDefinidaDecisaoJudicial")]
        public string AbrangenciaDefinidaDecisaoJudicial { get; set; }

        [JsonPropertyName("dataFimSancao")]
        public string DataFimSancao { get; set; }

        [JsonPropertyName("dataInicioSancao")]
        public string DataInicioSancao { get; set; }

        [JsonPropertyName("dataOrigemInformacao")]
        public string DataOrigemInformacao { get; set; }

        [JsonPropertyName("dataPublicacaoSancao")]
        public string DataPublicacaoSancao { get; set; }

        [JsonPropertyName("dataReferencia")]
        public string DataReferencia { get; set; }

        [JsonPropertyName("dataTransitadoJulgado")]
        public string DataTransitadoJulgado { get; set; }

        [JsonPropertyName("detalhamentoPublicacao")]
        public string DetalhamentoPublicacao { get; set; }

        [JsonPropertyName("fonteSancao")]
        public FonteSancaoModel FonteSancao { get; set; }

        [JsonPropertyName("fundamentacao")]
        public List<FundamentacaoModel> Fundamentacao { get; set; }

        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("informacoesAdicionaisDoOrgaoSancionador")]
        public string InformacoesAdicionaisDoOrgaoSancionador { get; set; }

        [JsonPropertyName("linkPublicacao")]
        public string LinkPublicacao { get; set; }

        [JsonPropertyName("numeroProcesso")]
        public string NumeroProcesso { get; set; }

        [JsonPropertyName("orgaoSancionador")]
        public OrgaoSancionadorModel OrgaoSancionador { get; set; }

        [JsonPropertyName("pessoa")]
        public PessoaJuridicaModel Pessoa { get; set; }

        [JsonPropertyName("sancionado")]
        public SancionadoModel Sancionado { get; set; }

        [JsonPropertyName("textoPublicacao")]
        public string TextoPublicacao { get; set; }

        [JsonPropertyName("tipoSancao")]
        public TipoSancaoModel TipoSancao { get; set; }

        [JsonPropertyName("valorMulta")]
        public string ValorMulta { get; set; }
    }
}

File: frontend/src/app/components/consultas/portal-transparencia/models/cnep.Model.ts
import { TipoSancao } from './tipoSancao.Model';
import { FonteSancao } from "./fonteSancao.Model";
import { Fundamentacao } from "./fundamentacao.Model";
import { OrgaoSancionador } from "./orgaoSancionador.Model";
import { PessoaJuridica } from "./pessoaJuridica.Model";
import { Sancionado } from "./sancionado.Model";

export interface Cnep{
        abrangenciaDefinidaDecisaoJudicial : string;
        dataFimSancao: string;
        dataInicioSancao: string;
       dataOrigemInformacao: string;
        dataPublicacaoSancao: string
        dataReferencia: string;
        dataTransitadoJulgado: string
        detalhamentoPublicacao: string
        fonteSancao: FonteSancao;
        fundamentacao: Fundamentacao[];
        id: number;
        informacoesAdicionaisDoOrgaoSancionador: string;
        linkPublicacao: string;
        numeroProcesso: string
        orgaoSancionador: OrgaoSancionador;
        pessoa: PessoaJuridica;
        sancionado: Sancionado;
        textoPublicacao: string;
        tipoSancao: TipoSancao;
        valorMulta: string;
}
